src/components/mc/cuda/mc_cuda.c

```diff
@@ -139,20 +142,23 @@ static ucc_status_t ucc_mc_cuda_mem_alloc(ucc_mc_buffer_header_t **h_ptr,
     h->from_pool = 0;
     h->mt        = UCC_MEMORY_TYPE_CUDA;
     *h_ptr       = h;
-    mc_trace(&ucc_mc_cuda.super, "allocated %ld bytes with cudaMalloc", size);
+    mc_trace(&ucc_mc_cuda.super, "allocated %ld bytes with %s", size,
+             mt == UCC_MEMORY_TYPE_CUDA ? "cudaMalloc" : "cudaMallocManaged");
     return UCC_OK;
 }
 
 static ucc_status_t ucc_mc_cuda_mem_pool_alloc(ucc_mc_buffer_header_t **h_ptr,
-                                               size_t                   size)
+                                               size_t                   size,
+                                               ucc_memory_type_t        mt)
 {
     ucc_mc_buffer_header_t *h = NULL;
-    if (size <= MC_CUDA_CONFIG->mpool_elem_size) {
+    if (size <= MC_CUDA_CONFIG->mpool_elem_size &&
+        mt != UCC_MEMORY_TYPE_CUDA_MANAGED) {
```

Comment:
If I understand correctly we choose not to allocate a mpool for cuda managed ?

Reply:
correct